File: golang/342_is_power_of_four/is_power_of_four.go
// Package main ...
package main

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func isPowerOfFour(n int) bool {
	return n > 0 && n&(n-1) == 0 && n%3 == 1
}

func main() {
	tests := []struct {
		n   int
		ans bool
	}{
		{16, true},
		{5, false},
		{1, true},
	}

	for index, test := range tests {
		assert.Equal(&testing.T{}, test.ans, isPowerOfFour(test.n), index)
	}
}

File: cpp/342_is_power_of_four/is_power_of_four.cpp
#include <cassert>
#include <tuple>
#include <vector>

class Solution {
 public:
  bool isPowerOfFour(int n) { return n > 0 && (n & (n - 1)) == 0 && n % 3 == 1; }
};

int main() {
  std::vector<std::tuple<int, bool>> tests{
      {16, true},
      {5, false},
      {1, true},
  };

  for (auto& [n, ans] : tests) {
    assert(Solution{}.isPowerOfFour(n) == ans);
  }
}
